upath/tests/cases.py

```diff
@@ -113,6 +113,12 @@ def test_iterdir2(self, local_testdir):
         assert set(p.name for p in pl_iter) == set(u.name for u in up_iter)
         assert next(self.path.parent.iterdir()).exists()
 
+    def test_parents(self):
+        p = self.path.joinpath("folder1", "file1.txt")
+        assert p.is_file()
+        assert p.parents[0] == p.parent
+        assert p.parents[1] == p.parent.parent
```

Comment:
Perhaps add a more explicit test, to not rely on `.parent`?
```suggestion
        assert p.parents[1] == p.parent.parent
        assert p.parents[0].name == "folder"
```

Reply:
`.parents` is not using `.parent` so the test is comparing two distinct implementations.